src/lib/support/ScopedBuffer.h

```diff
@@ -142,11 +148,13 @@ class ScopedMemoryBuffer : public Impl::ScopedMemoryBufferBase<MemoryManagement>
     inline T * Get() { return static_cast<T *>(Base::Ptr()); }
     inline T & operator[](size_t index) { return Get()[index]; }
 
-    inline const T * Get() const { return static_cast<T *>(Base::Ptr()); }
+    inline const T * Get() const { return static_cast<const T *>(Base::Ptr()); }
```

Comment:
Why this change?  The Base::Ptr is non-const.  we are casting it to T*, and then returning it as a `const T*`, which are all valid operations....

Doing a single cast that both changes the type and the const-ness not ideal.

Reply:
in Base, there are two functions, one is  void * Ptr() { return mBuffer; }
   another is const void * Ptr() const { return mBuffer; }